packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1159,9 +1299,7 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
 
   @override
   void moveChildRenderObject(covariant RenderObject child, int slot) {
-    // TODO(ianh): At some point we should be better about noticing when a
-    // particular LocalKey changes slot, and handle moving the nodes around.
-    assert(false);
+    renderObject.move(child, after: _currentBeforeChild);
```

Comment:
Should there be 

assert(slot != null);
assert(_currentlyUpdatingChildIndex == slot);